src/test/java/org/opensearch/security/dlic/dlsfls/DlsTermLookupQueryTest.java

```diff
@@ -0,0 +1,612 @@
+
+	}
+
+	@Test
+	public void testSimpleSearch_AccessCodes_1337_42() throws Exception {
+
+		setup(new DynamicSecurityConfig().setConfig("securityconfig_tlq.yml")
+				.setSecurityInternalUsers("internal_users_tlq.yml").setSecurityRoles("roles_tlq.yml")
+				.setSecurityRolesMapping("roles_mapping_tlq.yml"));
+
+		HttpResponse response = rh.executeGetRequest("/tlqdocuments/_search?pretty",
+				encodeBasicHeader("tlq_1337_42", "password"));
+		Assert.assertEquals(200, response.getStatusCode());
+		XContentParser xcp = XContentType.JSON.xContent().createParser(NamedXContentRegistry.EMPTY,
+				LoggingDeprecationHandler.INSTANCE, response.getBody());
+		SearchResponse searchResponse = SearchResponse.fromXContent(xcp);
+
+		// 15 docs, all need to have either access code 1337 or 42
+		Assert.assertEquals(searchResponse.toString(), 15, searchResponse.getHits().getTotalHits().value);
+		// fields need to have 42 or 1337 access code
+		assertAccessCodesMatch(searchResponse.getHits().getHits(), new Integer[] { 42, 1337 });
+
+	}
+
+	@Test
+	public void testSimpleSearch_AccessCodes_999() throws Exception {
+
+		setup(new DynamicSecurityConfig().setConfig("securityconfig_tlq.yml")
+				.setSecurityInternalUsers("internal_users_tlq.yml").setSecurityRoles("roles_tlq.yml")
+				.setSecurityRolesMapping("roles_mapping_tlq.yml"));
+
+		HttpResponse response = rh.executeGetRequest("/tlqdocuments/_search?pretty",
+				encodeBasicHeader("tlq_999", "password"));
+		Assert.assertEquals(200, response.getStatusCode());
+		XContentParser xcp = XContentType.JSON.xContent().createParser(NamedXContentRegistry.EMPTY,
+				LoggingDeprecationHandler.INSTANCE, response.getBody());
+		SearchResponse searchResponse = SearchResponse.fromXContent(xcp);
+
+		Assert.assertEquals(searchResponse.toString(), 0, searchResponse.getHits().getTotalHits().value);
+	}
+
+	@Test
+	public void testSimpleSearch_AccessCodes_emptyAccessCodes() throws Exception {
+
+		setup(new DynamicSecurityConfig().setConfig("securityconfig_tlq.yml")
+				.setSecurityInternalUsers("internal_users_tlq.yml").setSecurityRoles("roles_tlq.yml")
+				.setSecurityRolesMapping("roles_mapping_tlq.yml"));
+		SearchResponse searchResponse = executeSearch("tlqdocuments", "tlq_empty_access_codes", "password");
+		Assert.assertEquals(searchResponse.toString(), 0, searchResponse.getHits().getTotalHits().value);
+	}
+
+	@Test
+	public void testSimpleSearch_AccessCodes_noAccessCodes() throws Exception {
+
+		setup(new DynamicSecurityConfig().setConfig("securityconfig_tlq.yml")
+				.setSecurityInternalUsers("internal_users_tlq.yml").setSecurityRoles("roles_tlq.yml")
+				.setSecurityRolesMapping("roles_mapping_tlq.yml"));
+		SearchResponse searchResponse = executeSearch("tlqdocuments", "tlq_no_codes", "password");
+
+		Assert.assertEquals(searchResponse.toString(), 0, searchResponse.getHits().getTotalHits().value);
+	}
+
+	@Test
+	// TODO: Isn't that the same as above???
```

Comment:
??

Reply:
Sorry, this was a leftover from internal review, I will remove this test since it is simply a duplicate